woocommerce/woocommerce-android
# AI Product Description: Highlight Tooltips UI

Part of: #9290


### Description

This PR adds the UI for the highlight tooltips used for the "Write with AI" button in product descriptions.

It doesn't handle any functionality yet, just the design. The available design shown below is only on iOS in light mode, so I'm taking some liberties to convert some items into the Android equivalent. 

<img src="https://github.com/woocommerce/woocommerce-android/assets/266376/1505ff85-5b4e-4e01-b16c-1d03ad4e7862" width="42%" />

Here's the end result for Android.

| Light mode | Dark mode |
|--------|--------|
| ![Screenshot_20230705_201434](https://github.com/woocommerce/woocommerce-android/assets/266376/65b42fde-1cce-47c3-a2b8-7cba1bb2c036) | ![Screenshot_20230705_201418](https://github.com/woocommerce/woocommerce-android/assets/266376/58bfa2d2-57b9-4bfb-b7ff-5eda047e42cb) | 





### Testing instructions


1. Run app,
2. Go to product lists, select a product,
3. Make sure the highlights tooltip show the design as above.
4. Check that tapping outside of the highlight, or tapping the Got It button, will remove the tooltip.


### Known issues
The UI uses the `PopupWindow` class that comes with Android. In landscape mode it will look odd because the AI button as its anchor is likely not shown yet due to screen height. Ideally to fix that issue, the tooltip should appear as the user scrolls up. However, a `PopupWindow`'s position is set when we call `showAsDropDown(anchorView)` and it doesn't track the anchor view's position. If the anchor view moves because of scrolling, the `PopupWindow` will stay at its original position.

To make things more complicated, the behavior that makes `PopupWindow` be dismissed during tap outside of it, also dismisses it during scroll.

I think there's a way around those issues, but it feels like it'll add development time to handle an uncommon scenario. I'm happy to handle it in a future PR if we get a lot of user feedbacks about it, though. For now, this works fine in portrait mode.
